lib/geared_pagination/recordset.rb

```diff
@@ -32,7 +32,7 @@ def page_count
     end
 
     def records_count
-      @records_count ||= records.unscope(:limit).unscope(:offset).unscope(:select).count
+      @records_count ||= records.unscope(:limit).unscope(:offset).unscope(:select).to_a.size
```

Comment:
I think, this is very inefficient: imagine a model with millions of records... `to_a` would load them all. This feels like a hack, but should a little bit better:
```
    def records_count
      if @records_count.nil?
        @records_count = records.unscope(:limit).unscope(:offset).unscope(:select).count
        # records count could be a hash if query has a group by clause -> in this case records count is number of groups
        @records_count = @records_count.size if @records_count.is_a?(Hash)
      end

      @records_count
    end
```

Reply:
Great callout! I've updated the code and PR description.